deploy/kubernetes/docker/Dockerfile

```diff
@@ -0,0 +1,29 @@
+FROM openjdk:8-jdk-slim
```

Comment:
I suggest to use other general-purpose images, such as alpine or centos. 
Many companies may have their own JDK choices, so we should treat JDK as 3rd party lib hadoop.

Reply:
It's ok to use openjdk. You can see Spark's Dockerfile https://github.com/apache/spark/blob/b481ed36d129982bdd5aedd0d5864dd954dad006/resource-managers/kubernetes/docker/src/main/dockerfiles/spark/Dockerfile#L19

Reply:
I think we can use dependency_cache to save our build time

for example:
FROM openjdk:8-jdk-slim as dependency_cache

Reply:
Seems `8-jdk-slim` image is based on Debian stable, which is fine.
Can we use `8-jre-slim` here?

Reply:
> Seems `8-jdk-slim` image is based on Debian stable, which is fine. Can we use `8-jre-slim` here?

 What's your reason?

Reply:
> > Seems `8-jdk-slim` image is based on Debian stable, which is fine. Can we use `8-jre-slim` here?
> 
> What's your reason?

It's smaller.

Reply:
> I think we can use dependency_cache to save our build time
> 
> for example: FROM openjdk:8-jdk-slim as dependency_cache

It's great. Could you raise a follow-up pr which uses dependency_cache? If this pr don't have big problems, let's merge this pr quikly.

Reply:
> > > Seems `8-jdk-slim` image is based on Debian stable, which is fine. Can we use `8-jre-slim` here?
> > 
> > 
> > What's your reason?
> 
> It's smaller.

It may need more tests. If `8-jdk-slim` is fine, we can choose it first. When our e2e test is ready, wen can use e2e test to verify the `8-jre-slim`.